langchain/retrievers/azure_cognitive_search.py

```diff
@@ -64,16 +80,16 @@ def _headers(self) -> Dict[str, str]:
             "api-key": self.api_key,
         }
 
-    def _search(self, query: str) -> List[dict]:
-        search_url = self._build_search_url(query)
+    def _search(self, query: str, top_k: int) -> List[dict]:
```

Comment:
I don't think we need to pass `top_k` around, as it's only used in `_build_search_url` as is available there, because it's stored in `self`. So would delete this param.